src/mon/OSDMonitor.cc

```diff
@@ -78,9 +78,19 @@
 
 #include "json_spirit/json_spirit_reader.h"
 
+#include <boost/algorithm/string/predicate.hpp>
+
 #define dout_subsys ceph_subsys_mon
 #define OSD_PG_CREATING_PREFIX "osd_pg_creating"
 
+namespace {
+
+const uint32_t MAX_POOL_APPLICATIONS = 4;
```

Comment:
We're starting to get more prescriptive about sane defaults (like the max object size); here we just don't see a good reason for people to mix usage on a pool. If they're doing a mix of custom librados apps that's fine; they can give them the same name. But mixing cephfs and rgw is a good way to confuse recovery tools and have hard-to-manage performance issues, so let's discourage it as much as possible!

In particular, each app increases the size of the OSDMap. You don't get to inflate it as much as you want.